enhancements/security/manual-cloud-credentials.md

```diff
@@ -0,0 +1,170 @@
+---
+title: manual-cloud-credentials
+authors:
+  - "@dgoodwin"
+reviewers:
+  - @joelddiaz
+  - @smarterclayton
+  - @abhinavdahiya
+  - @derekwaynecarr
+  - @soltysh
+approvers:
+  - @joelddiaz
+  - @abhinavdahiya
+  - @soltysh
+creation-date: 2020-05-14
+last-updated: 2020-05-14
+status: provisional
+---
+
+# Support Manual Provisioning of Cloud Credentials
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This proposal outlines a plan for fully supporting the pre-existing process for disabling the cloud-credential-operator (still running, but in a no-op mode) and manually provisioning the cloud credentials for your cluster. It allows users to avoid broken upgrades when credentials have changed, and provides them with the tools to detect and reconcile the problem.
+
+## Motivation
+
+The current best practice for installing OpenShift 4 involves providing openshift-install with `root` cloud credentials powerful enough to mint additional fine grained credentials for each component in the cluster that needs them. The root credentials are stored in a secret in the cluster. This mechanism allows each component to have only the permissions it needs, and the required permissions are defined in the release image and automatically upgraded with the cluster itself.
+
+However some customers are unable or unwilling to do this and would prefer to disable automatic credential provisioning, and instead manage it themselves. It is possible to disable the cloud-credential-operator today by injecting a `ConfigMap` into the openshift-install manifests directory, along with several `Secrets` containing the cloud provider credentials the user manually configured. While this process exists today, it is not officially supported as we believe there are situations that could arise during upgrade where credentials have changed, and the upgrade will block and go into a failed state, with little visibility as to why or how it can be fixed.
+
+### Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+
+Goals for users who have chosen to disable the cloud credential operator:
+
+  * Users can easily mint and inject their own cloud credential secrets at install time.
+  * Users will be prevented from upgrading *if* credentials have changed in an available release image update.
+  * Users can upgrade without failures when the new release image contains credential changes.
```

Comment:
hmm.. i'm not sure I completely understand the difference between these 2 goals.

Reply:
Mmm yeah they are the same, will drop one.

Reply:
Should it be "users can upgrade without failures when the new release image contains *no* credentials changes."?